### PR TITLE
passwordverification

### PR DESCRIPTION
**Pull request by Richboy12345**

### CODE REVIEW

File: src/main/resources/templates/authentication/register.html [hunk 1]
```diff
@@ -19,7 +19,10 @@
 
 <!DOCTYPE html>
 <html lang="en" xmlns:th="http://www.thymeleaf.org">
-<head th:insert='parts/head.html :: head'></head>
+<head th:insert='parts/head.html :: head'>
+    <script defer src = "parentPassVerify.js"></script>
+    <script defer src = "studentPassVerify.js"></script>
```
Comment (varun-patel) [inline src/main/resources/templates/authentication/register.html:24]: please absolute link from root

File: src/main/resources/templates/authentication/register.html [hunk 2]
```diff
@@ -19,7 +19,10 @@
 
 <!DOCTYPE html>
 <html lang="en" xmlns:th="http://www.thymeleaf.org">
-<head th:insert='parts/head.html :: head'></head>
+<head th:insert='parts/head.html :: head'>
```
Comment (varun-patel) [inline src/main/resources/templates/authentication/register.html:22]: please move imports of scripts to after </body> before/after th:replace='parts/scripts :: scripts'

File: src/main/resources/templates/authentication/register.html [hunk 3]
```diff
@@ -41,7 +44,8 @@
                      aria-labelledby="nav-student-tab">
                     <br>
                     <h3>Student Registration</h3>
-                    <form th:object="${user}" method="post" action="/register/student">
+                    <form id="student-form" th:object="${user}" method="post" action="/register/student">
+                        <div id = "student-error"></div>
```
Comment (varun-patel) [inline src/main/resources/templates/authentication/register.html:48]: @ThePupoMaster please format this into a warning/danger bootstrap alert

File: src/main/resources/templates/authentication/register.html [hunk 4]
```diff
@@ -139,7 +143,8 @@ <h3>Student Registration</h3>
                 <div id="nav-parent" class="tab-pane fade" role="tabpanel" aria-labelledby="nav-parent-tab">
                     <br>
                     <h3>Parent Registration</h3>
-                    <form th:object="${user}" method="post" action="/register/parent">
+                    <form id="parent-form" th:object="${user}" method="post" action="/register/parent">
+                        <div id = "parent-error"></div>
```
Comment (varun-patel) [inline src/main/resources/templates/authentication/register.html:147]: @ThePupoMaster please format this into a warning/danger bootstrap alert

File: src/main/resources/templates/authentication/studentPassVerify.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    /*let validPassphrase = passphraseExpression.test(String(passphrase.value))
+    if (!validPassphrase) {
+        messages.push('Invalid Passphrase. Passphrase must be between 20 to 50 characters and contain only alphanumeric characters.')
+    }*/
+
+    if (messages.length > 0) {
+        e.preventDefault()
+        errorElement.innerText = messages.join(' ')
+    }
+})
```
Comment (varun-patel) [inline src/main/resources/templates/authentication/studentPassVerify.js:28]: move to /src/main/resources/static/js

File: src/main/resources/templates/authentication/parentPassVerify.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    /*let validPassphrase = passphraseExpression.test(String(passphrase.value))
+    if (!validPassphrase) {
+        messages.push('Invalid Passphrase. Passphrase must be between 20 to 50 characters and contain only alphanumeric characters.')
+    }*/
+
+    if (messages.length > 0) {
+        e.preventDefault()
+        errorElement.innerText = messages.join(', ')
+    }
+})
```
Comment (varun-patel) [inline src/main/resources/templates/authentication/parentPassVerify.js:28]: move to /src/main/resources/static/js

**Review by varun-patel — CHANGES_REQUESTED** [review]
Please review thymeleaf templating as well as speak with @ThePupoMaster about formatting alerts.
Otherwise looks good.

File: src/main/resources/templates/authentication/register.html (hunk 3 above)
Comment (andrewhuhh) [reply src/main/resources/templates/authentication/register.html:48]: https://v5.getbootstrap.com/docs/5.0/components/alerts/

**Comment by varun-patel** [comment]
This pull request is being marked as inactive and will be closed.

**Pull request closed without merging**